pkgs/build-support/rust/replace-workspace-values.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import toml
+
+crate_manifest = toml.load(sys.argv[1])
+workspace_manifest = toml.load(sys.argv[2])["workspace"]
+
+
+def replace_key(table, section, key):
+    if "workspace" in table[key] and table[key]["workspace"] is True:
+        print("replacing " + key)
+
+        replaced = table[key]
+        del replaced["workspace"]
+
+        workspace_copy = workspace_manifest[section][key]
+
+        if section == "dependencies":
+            crate_features = replaced.get("features")
+
+            if type(workspace_copy) is str:
+                replaced["version"] = workspace_copy
+            else:
+                replaced.update(workspace_copy)
+
+            merged_features = (crate_features or []) + (workspace_copy.get("features") or [])
+
+            if len(merged_features) > 0:
+                # Dictionaries are guaranteed to be ordered (https://stackoverflow.com/a/7961425)
+                replaced["features"] = list(dict.fromkeys(merged_features))
+        elif section == "package":
+            table[key] = replaced = workspace_copy
+
+
+def replace_dependencies(root):
+    for key in ["dependencies", "dev-dependencies", "build-dependencies"]:
+        if key in root:
+            for k, v in root[key].items():
```

Comment:
```suggestion
            for k in root[key].keys():
```

Reply:
Oops, result of a refactor and I didn't change that line. Surprised flake8 didn't warn about it...